### ISSUE TITLE
Bse run with ```BSEprop= "photolum"``` crashes with array mismatch

### ISSUE THREAD

**Issue by sangallidavide** [yambo-code/yambo#116]
Bse run (same inputs) with ```BSEprop= "photolum"``` crashed with array mismatch between ```BS_T_grp(i_T_g)%dipoles_opt(:,i_T,1)``` and ```PL_weights(:)``` in file https://github.com/yambo-code/yambo/blob/master/src/bse/K_IP.F#L188
This seems like a valid bug as printing their sizes gave me different values (1 and 3) respectively.
also in https://github.com/yambo-code/yambo/blob/master/src/bse/K_components_folded_in_serial_arrays.F#L111

One more
In https://github.com/yambo-code/yambo/blob/master/src/bse/PL_diago_residual.F ,the shape of BSS_dipoles_PL is assumed to be ```(BS_H_dim,3)```, but its allocated array shape is different ```(3,BS_H_dim)``` in
https://github.com/yambo-code/yambo/blob/master/src/modules/mod_PHOTOLUM.F#L47
Again in https://github.com/yambo-code/yambo/blob/master/src/bse/K_components_folded_in_serial_arrays.F#L111
the shape is assumed to be ```(3,BS_H_dim)```. This is an undefined behaviour, not sure how the test cases passed in this case

Size mismatch error : 
https://github.com/yambo-code/yambo/blob/master/src/bse/PL_diago_residual.F#L126

BSE inputfile : [bse.txt](https://github.com/user-attachments/files/16370926/bse.txt)

_Originally posted by @muralidhar-nalabothula in https://github.com/yambo-code/yambo/issues/111#issuecomment-2249351535_

**Comment by sangallidavide** [comment]
The PL tests in the test-suite are tagged as  `BROKEN`

**Comment by sangallidavide** [comment]
This was fixed in this branch: https://github.com/sangallidavide/yambo/tree/fixes-photoluminescence

**Comment by muralidhar-nalabothula** [comment]
@sangallidavide Looking at the changes, I see that K_diago_residuals are changed. Could you please tell me what changed  compared to the previous version. I am asking this because, I am about the complete the Ydiago interface and rewrote some of these functions to take care of parallel distribution of eigen-vectors . Here is my new implementation : https://github.com/muralidhar-nalabothula/yambo/blob/master/src/bse/K_residuals.F

**Comment by sangallidavide** [comment]
Yeah, I recently merge in tech master with changes done some time ago. Sorry it overlapped with your development on distributed residuals.

In the previous version:
- there were different "diago residuals" subroutine for each observable (absorption, kerr, magnons), with two different versions, one hermitian (K_diago_hermitian_residuals), and one non-hermitian (K_diago_non_hermitian_residuals). 6 subrotines in total
- both subroutines computed both the left and the right residuals

In the new version:
- there are two subroutines, one for the left and one for the right residuals with options for the kind of observable (abs, kerr, magnons).
- the one for the right residuals is always identical 
- the one for the left residuals was coded with two modes, one for the kind of scheme "hermitian" (e.g. TDA) and one for the kind scheme "non-hermitian (e.g. with coupling). Looking at the code now, I think it is always used in coupling mode. So I think the code could be cleaned a bit.
